DoSomething/gambit-conversations
# Macro cleanup + Profile updates part 2 [pr]

#### What's this PR do?

Continues part 1 (#408) by consolidating helper configs that define Rivescript macro behavior into `config/lib/macro` (the message a macro should send - if hardcoded, or the user fields it should update if required)

* Adds a hardcoded Rivescript trigger and topic to ask user for voting plan status, and updates profile when a valid response has been submitted.

* Deprecates the `support-requested` middleware by setting the topic within hardcoded Rivescript (the middleware predates the ability to support for a topic change via macro and Rivescript syntax)

#### How should this be reviewed?

* Test the various responses to the `make a plan` trigger and verify your `voting_plan_status` is updated (refs https://github.com/DoSomething/gambit-admin/pull/73)

* Verify that `question` trigger works as expected: changes user topic to `support` and sends the `supportRequested` template.

* Verify subscription status triggers like JOIN, LESS, STOP work as expected: correct templates are sent and user `sms_status` is updated accordingly

#### Any background context you want to provide?

A sweet vLater feature could be a new  `macroText` content type in Contentful, that has a single-value select field of macro names, and a long text -- used to override the macro's hardcoded `text`.

#### Relevant tickets
Refs https://www.pivotaltracker.com/story/show/160824069
Fixes https://www.pivotaltracker.com/story/show/159529212

#### Checklist
